### PR TITLE
Add S2I Action

### PR DESCRIPTION
**Pull request by jacobsee**
Adds the first GitHub Action (for building container images with s2i) to the repo!

### CODE REVIEW

File: s2i/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (tylerauerbeck) [inline s2i/LICENSE]: @jacobsee I think most of the CoP stuff uses the Apache license. We probably want to 1) bump this up to the root of the repo and 2) switch this just to stay consistent.

File: s2i/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+FROM docker:19.03.2
```
Comment (tylerauerbeck) [inline s2i/Dockerfile]: Not a blocker, but more of a question. Is there any reason why we're avoiding using a ubi image as the base?

File: s2i/LICENSE (hunk 1 above)
Comment (jacobsee) [reply s2i/LICENSE]: Is it possible in this scenario? This repo includes code originally from vrutkovs/action-s2i, which is MIT licensed...

File: s2i/Dockerfile (hunk 2 above)
Comment (jacobsee) [reply s2i/Dockerfile]: The s2i binary is pretty tightly integrated with Docker (see https://github.com/openshift/source-to-image/issues/966). I would love to see a UBI/Podman approach supported by that project, but until then, I think we need to use Docker - and I wasn't aware of a UBI that had actual Docker already installed & didn't necessarily want to install it onto a UBI at action-build time...

Comment (tylerauerbeck) [reply s2i/Dockerfile]: Sounds good. Possibly something to look into later then.

File: s2i/LICENSE (hunk 1 above)
Comment (tylerauerbeck) [reply s2i/LICENSE]: I haven't heard any feedback so I think we'll just move forward with having the Apache license at the root of the repo, but continuing to include the MIT license within this s2i directory unless you have any objections.

Comment (jacobsee) [reply s2i/LICENSE]: Nope, that sounds perfect to me!

**Pull request merged**